packages/ui/src/ui/pages/query-tracker/module/query/actions.ts

```diff
@@ -195,3 +209,34 @@ export function resetQueryTracker(): ThunkAction<any, RootState, any, never> {
         dispatch(createEmptyQuery());
     };
 }
+
+export function setQueryACO({
+    aco,
+    query_id,
+}: {
+    aco: string;
+    query_id: string;
+}): ThunkAction<Promise<unknown>, RootState, any, AnyAction> {
+    return (dispatch) => {
+        return wrapApiPromiseByToaster(dispatch(updateACOQuery({query_id, aco})), {
+            toasterName: 'update_aco_query',
+            skipSuccessToast: true,
+            errorTitle: 'Failed to update query ACO',
+        }).then(() => {
+            dispatch({type: UPDATE_ACO_QUERY, data: aco});
+            dispatch({type: UPDATE_DRAFT_ACO_QUERY, data: aco});
```

Comment:
All modification can be done with one UPDATE_ACO_QUERY action.

Reply:
There are two cases.
1: When making changes to the ACO for an existing query, both the draft and the query must be updated.
2: Update only the draft without modifying the existing query.